swift/apple/FirezoneKit/Sources/FirezoneKit/Views/GrantNotificationsView.swift

```diff
@@ -23,21 +23,17 @@ public final class GrantNotificationsViewModel: ObservableObject {
   }
 
   func grantNotificationButtonTapped(errorHandler: GlobalErrorHandler) {
-    Task.detached { [weak self] in
-      guard let self else { return }
-
+    Task {
       do {
         let decision = try await sessionNotification.askUserForNotificationPermissions()
         await onDecisionChanged(decision)
       } catch {
         Log.error(error)
 
-        await MainActor.run {
-          errorHandler.handle(ErrorAlert(
-            title: "Error granting notifications",
-            error: error
-          ))
-        }
+        errorHandler.handle(ErrorAlert(
+          title: "Error granting notifications",
+          error: error
+        ))
```

Comment:
Won't alerts also block the UI thread until they are dismissed?

Reply:
`NSAlert` will, but this is only available on `macOS`. But in most cases that's what we want, and Sentry's app hang protection is smart enough to avoid alerting those most of the time.

These `ErrorAlert`s are for iOS only and used within SwiftUI which _do not_ block since they use an `.alert` attachment on another view object (like a sheet)